wasm/src/lib.rs

```diff
@@ -58,7 +58,20 @@ fn oom(layout: ::core::alloc::Layout) -> ! {
 
 #[no_mangle]
 extern "C" fn _iroha_wasm_alloc(len: usize) -> *const u8 {
-    core::mem::ManuallyDrop::new(Vec::<u8>::with_capacity(len)).as_mut_ptr()
+    if len == 0 {
+        debug::dbg_panic("Cannot allocate 0 bytes");
+    }
+    let layout = alloc::alloc::Layout::array::<u8>(len).dbg_expect("Cannot allocate layout");
+    // Safety: safe until `layout` has non-zero size
+    unsafe { alloc::alloc::alloc_zeroed(layout) }
+}
```

Comment:
Using `Vec` to allocate memory and then just `Box` to destroy this memory doesn't look right.
So I decided to replace it with this. This should take less place too.
See [here](https://www.reddit.com/r/rust/comments/jzwwqb/about_creating_a_boxed_slice/) for explanation

Reply:
wow, significant difference in size
do we need overhead of zeroing memory?

Reply:
I'm not sure. But `Vec` does this. I just wanted to keep API the same. That's probably another question to @mversic

Reply:
> I'm quite sure it doesn't. Stack part is initialized, but stuff on heap is uninitialized

Hm... When why does [this program](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e1e98efc7cd5e04565db471462ce9ce2) print slice of zeros?

Reply:
I may be wrong but UB can print anything

Reply:
> Hm... When why does [this program](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e1e98efc7cd5e04565db471462ce9ce2) print slice of zeros?

the fact that this program prints all 0s doesn't mean that data is initialized to 0. Uninitialized data can be 0 as well

Reply:
Int

> > I'm quite sure it doesn't. Stack part is initialized, but stuff on heap is uninitialized
> 
> Hm... When why does [this program](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e1e98efc7cd5e04565db471462ce9ce2) print slice of zeros?

Internally `Vec` rely on [`Allocator::allocate`](https://doc.rust-lang.org/std/alloc/trait.Allocator.html#tymethod.allocate) function which do not guarante that data would be zero-initialized.